src/TLuaInterpreter.cpp

```diff
@@ -5484,222 +5483,256 @@ int TLuaInterpreter::tempTimer(lua_State* L)
+    TLuaInterpreter* pLuaInterpreter = host.getLuaInterpreter();
     bool multiLine, matchAll, highlight, playSound, filter, colorTrigger;
     int fireLength, lineDelta;
     QString fgColor, bgColor;
     QStringList regexList;
-    QString script, parent, pattern, soundFile;
+    QString parent, script, triggerName, pattern, soundFile;
     QColor hlFgColor, hlBgColor;
+    QList<int> propertyList;
+    int triggerID;
+
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #1 type (trigger name create or add to as string expected, got %s!)", luaL_typename(L, 1));
         lua_error(L);
         return 1;
-    } else {
-        parent = lua_tostring(L, 1);
-    }
-
-    if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+    } else if (!lua_isstring(L, 2)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #2 type (regex pattern to match as string, got %s!)", luaL_typename(L, 2));
         lua_error(L);
         return 1;
-    } else {
-        pattern = lua_tostring(L, 2);
-    }
-    if (!lua_isstring(L, 3)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+    } else if (!lua_isstring(L, 3) && !lua_isfunction(L, 3)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #3 type (code to run as a string or a function expected, got %s!)", luaL_typename(L, 3));
         lua_error(L);
         return 1;
-    } else {
-        script = lua_tostring(L, 3);
-    }
-    if (!lua_isnumber(L, 4)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+    } else if (!lua_isnumber(L, 4)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #4 type (multiline flag as number expected, got %s!)", luaL_typename(L, 4));
+        lua_error(L);
+        return 1;
+    } else if (!lua_isnumber(L, 7)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #7 type (filter flag as number expected, got %s!)", luaL_typename(L, 7));
+        lua_error(L);
+        return 1;
+    } else if (!lua_isnumber(L, 8)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #8 type (match all flag as number expected, got %s!)", luaL_typename(L, 8));
+        lua_error(L);
+        return 1;
+    } else if (!lua_isnumber(L, 11)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #11 type (fire length as number expected, got %s!)", luaL_typename(L, 11));
+        lua_error(L);
+        return 1;
+    } else if (!lua_isnumber(L, 12)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #12 type (line delta as number expected, got %s!)", luaL_typename(L, 12));
         lua_error(L);
         return 1;
-    } else {
-        multiLine = lua_tonumber(L, 4);
     }
+
+    parent = lua_tostring(L, 1);
+    pattern = lua_tostring(L, 2);
```

Comment:
These two need to be `...= QString::fromUtf8(lua_tostring(L, `x`));` I think...

Reply:
Adjusted